src/app/client/src/app/modules/dashboard/components/report/report.component.ts

```diff
@@ -144,14 +161,18 @@ export class ReportComponent implements OnInit {
               map((apiResponse) => {
                 const [data, reportSummary] = apiResponse;
                 const result: any = Object.assign({});
-                result['charts'] = (charts && this.reportService.prepareChartData(charts, data, updatedDataSource,
+                this.charts = (charts && this.reportService.prepareChartData(charts, data, updatedDataSource,
                   _.get(reportConfig, 'reportLevelDataSourceId'))) || [];
+                result['charts'] = this.charts;
                 result['tables'] = (tables && this.reportService.prepareTableData(tables, data, _.get(reportConfig, 'downloadUrl'),
                   this.hash)) || [];
                 result['reportMetaData'] = reportConfig;
                 result['reportSummary'] = reportSummary;
                 result['files'] = this.reportService.getParameterizedFiles(files || [], this.hash);
                 result['lastUpdatedOn'] = this.reportService.getFormattedDate(this.reportService.getLatestLastModifiedOnDate(data));
+                this.reportResult = result;
```

Comment:
Why do you need four different variables to store the same value? Please remove the extra ones...

Reply:
And refactor the code in the component to use the single property...

Reply:
resolved